src/macaron/dependency_analyzer/java_repo_finder.py

```diff
@@ -171,24 +199,52 @@ def find_repo(gav: str, tags: list[str]) -> Iterator[str]:
         "repofinder.java", "artifact_repositories", fallback=["https://repo.maven.apache.org/maven2"]
     )
 
-    # Parse the GAV and create the URLs for its POM
-    group, artifact, version = parse_gav(gav)
-    if not group:
-        return iter([])
-    request_urls = create_urls(group, artifact, version, repositories)
-    if not request_urls:
-        return iter([])
-
-    # Try each POM URL in order, terminating early if a match is found
-    with requests.Session() as session:
-        pom = ""
-        for request_url in request_urls:
-            pom = retrieve_pom(session, request_url)
-            if pom != "":
-                break
-
-    if pom == "":
-        return iter([])
-
-    # Parse XML data and return URL
-    return parse_pom(pom, tags)
+    # Perform the following in a loop:
+    # - Create URLs for the current artifact POM
+    # - Parse the POM
+    # - Try to extract SCM metadata and return URLs
+    # - Try to extract parent information and change current artifact to it
+    # - Repeat
+    limit = 10
```

Comment:
@behnazh-w  I wonder if we should put this variable into `defaults.ini`?

Reply:
> @behnazh-w I wonder if we should put this variable into `defaults.ini`?

Yes it should.